### PR TITLE
[Merged by Bors] - refactor: rename `FreeProduct` to `Monoid.CoprodI`

### PR DESCRIPTION
**Pull request by urkud**
---
Should it be `CoprodI` or `ICoprod`?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ocfnash** [comment]
What is the motivation for this renaming?

Would moving `FreeProduct` into the `Monoid` namespace achieve the same aims?

**Comment by urkud** [comment]
> What is the motivation for this renaming?
> 
> Would moving `FreeProduct` into the `Monoid` namespace achieve the same aims?

See leanprover-community/mathlib#18437.
I'm going to add a `Prod` version and in that mathlib 3 PR I was told to use `Coprod`/`Coprodᵢ` (it was before we renamed `Unionᵢ` to `iUnion`).

File: Mathlib/GroupTheory/CoprodI.lean [hunk 1]
```diff
@@ -13,52 +13,66 @@ import Mathlib.Data.Set.Pointwise.SMul
...
 
-When `M i` are all groups, `FreeProduct M` is also a group (and the coproduct in the category of
-groups).
+When `ι` and all `M i` have decidable equality,
+the free product bijects with the type `Monoid.CoprodI.Word M` of reduced words.
+This bijection is constructed
+by defining an action of `Monoid.CoprodI M` on `Monoid.CoprodI.Word M`.
+
+When `M i` are all groups, `Monoid.CoprodI M` is also a group
+(and the coproduct in the category of groups).
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/GroupTheory/CoprodI.lean:29]: This change is motivated by [this discussion](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Metaprogramming.20tutorial/near/374887296)

[the remark this answers is not in the document]
Comment (ocfnash) [reply Mathlib/GroupTheory/CoprodI.lean:29]: I see thanks, this idea is new to me. I'm not sure if I like it but certainly the unanimous vote is in your favour!

**Pull request closed without merging**